spec/unit/shoulda/matchers/active_record/have_rich_text_matcher_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'unit_spec_helper'
+
+describe Shoulda::Matchers::ActiveRecord::HaveRichText, type: :model do
+  def self.rich_text_is_defined?
+    defined?(ActionText::RichText)
+  end
+
+  if rich_text_is_defined?
+    context 'when the model has a RichText association' do
+      it 'matches when the subject configures has_rich_text' do
+        valid_record = new_post(is_rich_text_association: true)
+
+        expected_message = 'Did not expect Post to have ActionText::RichText :content'
+
+        expect { have_rich_text(:content) }.
+          to match_against(valid_record).
+          or_fail_with(expected_message)
+      end
+
+      it 'returns the message including the name of the provided association' do
+        matcher = have_rich_text(:content)
+        expect(matcher.description).
+          to eq('have configured :content as a ActionText::RichText association')
+      end
+    end
+
+    context 'when the model does not have a RichText association' do
+      it 'does not match when provided with a model attribute' do
+        invalid_record = new_post(has_invalid_content: true)
+        expected_message = 'Expected Post to have configured :invalid_content as a ' \
+          'ActionText::RichText association but :invalid_content is a NilClass'
+
+        expect { have_rich_text(:invalid_content) }.
+          not_to match_against(invalid_record).
+          and_fail_with(expected_message)
+      end
+
+      it 'does not match when lacking has_rich_text attribute' do
+        invalid_record = new_post
+        expected_message = 'Expected Post to have configured :invalid_attribute as a ' \
+          'ActionText::RichText association but :invalid_attribute does not exist'
+
+        expect { have_rich_text(:invalid_attribute) }.
+          not_to match_against(invalid_record).
+          and_fail_with(expected_message)
+      end
+    end
+  else
+    it 'raises a undefined method error when ActionText is not defined' do
```

Comment:
Sort of a similar thing here — for _both_ of these tests, not just this one, ActionText is not defined (that's why you have them in an `else`). Can you have the names reflect the fact that in one, the attribute being tested exists, whereas in the other it does not?

Reply:
Done!